Update ArchLinux compiling instructions

mrmurb opened the pull request:
Fixed references to non-existing package and wrongly pointed symlinks.

Tested and compiling with minor problems. CMake finds `qwt5-qt4` but for some reason it will not compile `odcockpit` by default. 
I "fixed" this by commenting out line 101-102 & 105-106 (the IF for Qt4 and QWT5QT4) in `CMakeLists.txt` to force CMake to create build instructions for `odcockpit`. 

This might not be a nice solution but I have little experience using CMake and I'm sure someone else can fix it.

mrmurb committed:
Update ArchLinux compiling instructions

Fixed package references and library links.

chrberger commented:
Thanks for the suggested updates for the documentation.

CMake uses these two scripts to find libqwt5qt4:
(1) https://github.com/se-research/OpenDaVINCI/blob/master/cmake/Modules/FindQwt5Qt4.cmake
(2) https://github.com/se-research/OpenDaVINCI/blob/master/odcockpit/cmake.Modules/FindQwt5Qt4.cmake

The first one for the entire build tree to determine whether to build odcockpit or not, and the second one for building odcockpit.

Could you please check if the following files that the script is looking for on your platform are at the specified locations:
- /usr/include/local/qwt-qt4/qwt.h or /usr/include/qwt-qt4/qwt.h
- /usr/lib/libqwt-qt4.so or /usr/lib64/libqwt-qt4.so

Thanks.

mrmurb commented:
The location of the files after creating the symlinks from the documentation page is:
- `qwt.h` in /usr/include/qwt-qt4
- `libqwt-qt4.so` in /usr lib

I looked in `FindQwt5Qt4.cmake` and saw the directories when doing this yesterday. Even though CMake finds the libraries as you can see down below it will not compile odcockpit from the get go.

We have now tried this on three separate computers and everything is working fine except from those needed changes to bypass the automated libqwt5qt4 find.

> -- Found CxxTest: Compiling and running test suites enabled.
> -- Found ant executable: /usr/bin/ant
> CMake Warning at cmake/Modules/CheckBuildPackages.cmake:24 (MESSAGE):
>   Could not find dpkg executable to create .deb packages.
> Call Stack (most recent call first):
>   cmake/Modules/CheckPlatform.cmake:36 (INCLUDE)
>   CMakeLists.txt:29 (INCLUDE)
> 
> CMake Warning at cmake/Modules/CheckBuildPackages.cmake:35 (MESSAGE):
>   Could not find rpmbuild executable to create .rpm packages.
> Call Stack (most recent call first):
>   cmake/Modules/CheckPlatform.cmake:36 (INCLUDE)
>   CMakeLists.txt:29 (INCLUDE)
> 
> -- Found librt: /usr/include, /usr/lib64/librt.so Using static library.
> -- Found OpenCV headers:   /usr/include/opencv;/usr/include
> -- Found OpenCV libraries: /usr/lib64/libopencv_calib3d.so;/usr/lib64/libopencv_contrib.so;/usr/lib64/libopencv_core.so;/usr/lib64/libopencv_features2d.so;/usr/lib64/libopencv_flann.so;/usr/lib64/libopencv_gpu.so;/usr/lib64/libopencv_highgui.so;/usr/lib64/libopencv_imgproc.so;/usr/lib64/libopencv_legacy.so;/usr/lib64/libopencv_ml.so;/usr/lib64/libopencv_objdetect.so;/usr/lib64/libopencv_video.so
> -- Boost version: 1.60.0
> -- Found qwt5-qt4 library: /usr/include/qwt-qt4, /lib/libqwt-qt4.so
> -- Found popt: /usr/include, /lib/libpopt.so
> 
> Platform configuration for building the complete source tree:
> -- Found working Java compiler and ant build tool: Building odDataStructureGenerator
> -- Using /opt/od/bin/odDataStructureGenerator-latest.jar to generate code from data structure models.
> -- Using /opt/od/bin/odCANDataStructureGenerator-latest.jar to generate code from CAN data structure models.
> -- Found C and C++ compiler:   Building opendavinci-lib
> -- Found C++ compiler:         Building opendavinci-odsupercomponent
> -- Found C++ compiler:         Building opendavinci-odtools
> -- Found C++ compiler:         Building opendavinci-automotivedata
> -- Found C++ compiler:         Building opendavinci-tutorials
> -- Found C++ compiler, OpenCV, OpenGL, GLUT, and Boost: Building libodsimulation
> -- Found C++ compiler, OpenCV, OpenGL, GLUT, and Boost: Building odsimulation
> -- Found C++ compiler, OpenCV, OpenGL, GLUT, and Boost: Building automotive/miniature
> -- Found popt-dev:             Building automotive/odcantools
> Test command: "/usr/bin/ctest --output-on-failure"
> -- Using toolchain:            /home/max/opendanvinci/cmake/default-Toolchain.cmake
> -- Installing artifacts to:    /opt/od
> 
> -- Configuring done
> -- Generating done
> -- Build files have been written to: /home/max/opendanvinci/build
- Edit: Needed to revert changes done to get a proper print

mrmurb committed:
Fixed cmake find for Qt4

We're looking for Qt4, not Qt.

mrmurb committed:
CMake improvements and Qwt5 include refractoring.

CMake should be able to locate Qwt5 on distributions that are not based
of Debian. Also updated how Qwt is refrenced when including headers to
suit the new directory paths.

mrmurb committed:
Python 2.7 specified

Make sure that we search for Python 2.7 to not conflict with systems
that default to Python 3.

mrmurb committed:
CMake and Documentation improvements

Tell CMake to count Qwt5-Qt4 as a system library to suppress warnings
generated from those headers.

Removed unncessesary symlinks from the ArchLinux documentation.

mrmurb commented:
CMake now finds both Qt4 and Qwt5-Qt4 libraries without any symlinking.
CMake also selects Python 2.7 by default, removing the need for it to be symlinked as `python`. #23

This should also work fine for other distributions that does not use Debian-like file structure. So documentation for other distributions might need an update if you decide to merge this pull request.

I have successfully built the complete project without any hiccups.

So Long, and thanks for all the fish!
- Max

chrberger commented:
Hi Max,

I read the proposed changes that look good to me. Our Jenkins will tell us the results after merging later. Could you please have a look at the remaining documentation files (*.rst) to check if there are unnecessary symlinks left before we merge?

Thanks,

   Christian

mrmurb committed:
Refractoring PythonInterp

Specifying PythonInterp 2.7 in all locations.

mrmurb committed:
Added more file paths

mrmurb committed:
Removed unncessesary symlinks

mrmurb commented:
I've refactored all the references to `PythonInterp` to `PythonInterp 2.7` in all occurrences.

Also added more search paths to `FindQwt5Qt4.cmake` to support paths accordingly to your old installation documentation. Removed all the unnecessary links from distributions I feel comfortable to change. So *BSD and other more unusual flavors of Linux are left untouched.

See you in class tomorrow (or maybe later at supervision)!
- Max

chrberger commented:
Thanks for the updates. Let's merge and let Jenkins check the integration.

mrmurb commented:
Great! Hopefully there will be no problems!
Fun to contribute :smiley: 

Accidentally restored the branch...

chrberger commented:
You can see Jenkin's results as build tags on this page: https://github.com/se-research/OpenDaVINCI

chrberger commented:
Jenkins is happy.

mrmurb commented:
Great! This would have solved #23 then.

chrberger commented:
v2.4.8 is out with your suggested changes: https://github.com/se-research/OpenDaVINCI/releases/tag/v2.4.8
